src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Refactoring/RenameEndpoint.cs

```diff
@@ -71,20 +70,28 @@ internal class RenameEndpoint : AbstractRazorDelegatingEndpoint<RenameParamsBrid
         protected override string EndpointName => LanguageServerConstants.RazorRenameEndpointName;
 
         /// <inheritdoc/>
-        protected override async Task<WorkspaceEdit?> HandleInRazorAsync(RenameParamsBridge request, CancellationToken cancellationToken)
+        protected override async Task<WorkspaceEdit?> TryHandleAsync(RenameParamsBridge request, CancellationToken cancellationToken)
         {
-            var documentContext = await _documentContextFactory.GetRequiredAsync(request.TextDocument.Uri, cancellationToken);
+            var documentContext = await _documentContextFactory.CreateAsync(request.TextDocument.Uri, cancellationToken);
```

Comment:
I think I missed this before, but I'm not clear on this. Two thoughts: Firstly, if every implementor of this method is going to have to get the document context, then the base class can just do it and pass it in.

Second though: When the base class gets the document context, if it failes it returns null. When this gets the document context, if it fails, it throws. We should be standardizing on an approach. I'm fine with throwing if we know the base class has done the check already, as then its more of a contract check, but at this point in the code that isn't true, and if it is true, then it adds more weight to passing the document context in as a parameter.

Reply:
I'm happy to add `documentContext` to TryHandleAsync as part of the contract, since the base already is getting it. We could also add some protected readonly type to wrap the contract in. If ever we need to add more... it's code style question at that point